tests/integration/ambient/registry_setup_test.go

```diff
@@ -35,11 +35,23 @@ const (
 )
 
 func testRegistrySetup(ctx resource.Context) (err error) {
-	registry, err = registryredirector.New(ctx, registryredirector.Config{
-		Cluster:        ctx.AllClusters().Default(),
-		TargetRegistry: "kind-registry:5000",
-		Scheme:         "http",
-	})
+	var config registryredirector.Config
+
+	isKind := ctx.Clusters().IsKindCluster()
+
+	// By default, for any platform, the test will pull the test image from public "gcr.io" registry.
+	// For "Kind" environment, it will pull images from the "kind-registry".
+	// "Kind" environment in IPv6 format, unable to properly resolve external images path, like "gcr.io"
+	// https://github.com/istio/istio/pull/51778
+	if isKind {
+		config = registryredirector.Config{
+			Cluster:        ctx.AllClusters().Default(),
+			TargetRegistry: "kind-registry:5000",
+			Scheme:         "http",
+		}
+	}
```

Comment:
Been a busy day... but I am thinking this through.. if it's not kind, your config is basically empty, but you proceed to configure docker, etc. with it anyway. The logic here doesn't make sense to me.